### PR TITLE
fix: prevent GitHub clones from capturing terminal input

### PR DESCRIPTION
**Pull request by MDGChamomile**
## Summary

- run GitHub clone commands without interactive Git, Git Credential Manager, or GitHub CLI prompts
- isolate clone commands in a POSIX process group and terminate the full process tree on timeout or cancellation (with `taskkill /T` on Windows)
- add regression coverage for non-interactive clone environments and descendant cleanup

## Problem

When `fetch_content` receives a missing, moved, or inaccessible GitHub repository URL, Git can open a credential prompt directly on `/dev/tty`. The prompt is hidden behind Pi's TUI and captures editor input and shortcuts.

The existing `execFile(..., { timeout })` only terminates the direct `git clone` process. Its `git remote-https` / `git-remote-https` descendants can survive, remain attached to the foreground terminal, and keep consuming input after the fetch has already fallen back or returned.

This change uses `spawn` so POSIX clones can run in their own process group, disables interactive credential paths, and terminates the complete clone process tree on timeout or abort.

## Verification

- `node --test test/github-extract.test.mjs` — 6/6 passed
- `npm run typecheck` — passed
- `npm run audit:runtime` — passed
- live missing/moved-repository reproduction returned in 324 ms with no new orphaned Git helper
- full `npm test` — 271/272 passed; the unrelated `failed password lookups are retried instead of cached` Chrome-cookie test fails in this Linux environment

<details open><summary><h3>Greptile Summary</h3></summary>

This PR fixes orphaned Git helper processes capturing terminal input when a clone times out or the repository is inaccessible. It replaces `execFile` (which only killed the direct child) with `spawn` plus an explicit process-group shutdown: SIGTERM to `-pid` followed by SIGKILL after a 3-second grace period on POSIX, and `taskkill /T /F` on Windows. Three environment variables (`GIT_TERMINAL_PROMPT`, `GCM_INTERACTIVE`, `GH_PROMPT_DISABLED`) suppress interactive credential prompts before the clone even starts.

- **`github-extract.ts`** — New `terminateProcessTree` helper and a fully refactored `execClone` that uses `spawn` with `detached: true` (POSIX) for process-group isolation, manual timeout/abort signal plumbing with a `settled` guard, and SIGTERM→SIGKILL escalation.
- **`test/github-extract.test.mjs`** — Two new regression tests: one verifying the non-interactive env vars reach the child process, and one verifying a SIGTERM-resistant helper is force-killed via SIGKILL escalation.
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

Safe to merge — the change is well-scoped, the new process-group shutdown logic is correct, and the regression tests cover both the non-interactive env-var path and the SIGTERM-resistant process-tree kill.

The refactor from execFile to spawn with explicit SIGTERM→SIGKILL escalation handles all the edge cases correctly: the settled guard prevents double-resolution, the abort/timeout paths are mutually exclusive (onAbort clears the timeout), and unref'd timers won't block the event loop. No logic errors or data-loss paths were found in the changed code.

**Files Needing Attention:** No files require special attention.
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| github-extract.ts | Core change: replaces execFile with spawn, adds terminateProcessTree with SIGTERM→SIGKILL escalation and Windows taskkill /T /F, suppresses terminal prompts via env vars. Logic is sound — settled guard prevents double resolution, abort/timeout coordination is correct, and forceKill timers are unref'd to avoid blocking the event loop. |
| test/github-extract.test.mjs | Two new regression tests: one checks non-interactive env vars are propagated to the clone subprocess; the other verifies that a SIGTERM-resistant process group (parent + helper both ignoring SIGTERM) is ultimately force-killed via SIGKILL. waitForProcessExit polling and /proc/stat zombie detection are well-handled. |
| CHANGELOG.md | Adds a concise entry under [Unreleased] describing the fix. No issues. |

</details>

<sub>Reviews (2): Last reviewed commit: ["fix: force-kill stalled clone process gr..."](https://github.com/nicobailon/pi-web-access/commit/6b35ea0d000e0a7d7a9668d10cb6c7ae595a8abc) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=49333266)</sub>

### CODE REVIEW

File: test/github-extract.test.mjs [hunk 1]
```diff
@@ -91,6 +124,113 @@ test("normalizeClonePath handles absolute paths without expansion", async () =>
...
+test("GitHub clone timeout terminates descendant processes", { skip: process.platform === "win32" }, async () => {
+	const root = await mkdtemp(join(tmpdir(), "pi-web-access-github-timeout-tree-"));
+	const agentDir = join(root, "agent-dir");
+	const binDir = join(root, "bin");
+	const helperPidFile = join(root, "helper.pid");
+	await mkdir(agentDir, { recursive: true });
+	await mkdir(binDir, { recursive: true });
+	await writeFile(
+		join(agentDir, "web-search.json"),
+		JSON.stringify({ githubClone: { clonePath: join(root, "repos"), cloneTimeoutSeconds: 0.1 } }),
```
[the remark this answers is not in the document]
Comment (MDGChamomile) [reply test/github-extract.test.mjs]: Addressed in 6b35ea0. The clone timeout in this regression test is now 0.5 seconds, giving the fake Node.js git process enough startup time before timeout cleanup begins.

File: github-extract.ts [hunk 2]
```diff
@@ -186,10 +186,45 @@ function cloneDir(config: GitHubCloneConfig, owner: string, repo: string, ref?:
...
+		return;
+	}
+
+	try {
+		// Clone commands run in their own process group so git/gh helpers cannot
+		// survive a timeout or cancellation and keep reading from the host TTY.
+		process.kill(-pid, "SIGTERM");
+	} catch {
+		child.kill();
+	}
```
[the remark this answers is not in the document]
Comment (MDGChamomile) [reply github-extract.ts:214]: Addressed in 6b35ea0. POSIX process-tree termination now escalates to SIGKILL after a 3-second grace period and targets the same process group (`-pid`), not only the root child. The regression test makes both the fake git process and its helper ignore SIGTERM, then verifies that both exit via the fallback.

**Comment by nicobailon** [comment]
Thanks @MDGChamomile. I landed this via maintainer PR #207 so the non-interactive clone fix could be rebased cleanly with the current changelog and validation. Your contribution is preserved in the landed commit authorship and changelog credit, so I am closing this PR as superseded.

**Comment by nicobailon** [comment]
Superseded by merged maintainer PR #207.
